src/Runtime.Base/src/System/Runtime/RuntimeExports.cs

```diff
@@ -23,6 +23,19 @@ internal static class RuntimeExports
         public static unsafe object RhNewObject(EETypePtr pEEType)
         {
             EEType* ptrEEType = (EEType*)pEEType.ToPointer();
+
+            // This is structured in a funny way because at the present state of things in CoreRT, the Debug.Assert
+            // below will call into the assert defined in the class library (and not the MRT version of it). The one
+            // in the class library is not low level enough to be callable when GC statics are not initialized yet.
+            // Feel free to restructure once that's not a problem.
+            bool isValid = !ptrEEType->IsGenericTypeDefinition &&
```

Comment:
I doubt that the compiler will be able to remove all these checks in release. I think this should be under `#if DEBUG`.